src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -82,13 +82,13 @@ public function __construct()
             new ReplaceAliasByActualDefinitionPass(),
             new RemoveAbstractDefinitionsPass(),
             new RemoveUnusedDefinitionsPass(),
-            new InlineServiceDefinitionsPass(new AnalyzeServiceReferencesPass()),
             new AnalyzeServiceReferencesPass(),
+            new CheckExceptionOnInvalidReferenceBehaviorPass(),
+            new InlineServiceDefinitionsPass(new AnalyzeServiceReferencesPass()),
```

Comment:
`new AnalyzeServiceReferencesPass()` should still be listed after, as it was before, because `InlineServiceDefinitionsPass` changes the graph

Reply:
but `CheckExceptionOnInvalidReferenceBehaviorPass` can not be listed after `InlineServiceDefinitionsPass`, because this is the reason why current bug occurs

Reply:
`AnalyzeServiceReferencesPass` should be listed twice